server/bin/index-pbench

```diff
@@ -1215,39 +1500,223 @@ class ToolData(PbenchData):
         ...
         keyN: 100.N
         timestamp: ...
+
+        """
+        record = None
+        prev_gauge = None
+        ts_orig = None
+        prev_ts_orig = None
+        try:
+            remaps = self._remaps[self.toolname]
+        except KeyError:
+            remaps = None
+        idx = 0
+        for line in file_object:
+            if line.startswith('timestamp:'):
+                prev_ts_orig = ts_orig
+                if record:
+                    # timestamp delimits records, yield last record.
+                    if not record[self.toolname]['rate']:
+                        # For first record, rate will be empty, so
+                        # don't emit it.
+                        del record[self.toolname]['rate']
+                    yield record
+                    idx += 1
+                    # Be sure to remember the record we just emitted
+                    # so that it is available for rate calculations.
+                    prev_gauge = record[self.toolname]['gauge']
+                # Get the second column, the timestamp value, which is
+                # *seconds* since the epoch, and then convert to millis
+                # since the epoch.
+                ts_orig = float(line.split(':')[1])
+                if prev_ts_orig is not None:
+                    assert prev_ts_orig <= ts_orig, "prev_ts_orig %r > ts_orig %r" % (prev_ts_orig, ts_orig)
+                ts_str = self.mk_abs_timestamp_millis(ts_orig * 1000)
+                record = _dict_const()
+                record['@timestamp'] = ts_str
+                record['@timestamp_original'] = str(ts_orig)
+                record['run'] = self.run_metadata
+                record['iteration'] = self.iteration_metadata
+                record['sample'] = self.sample_metadata
+                record[self.toolname] = _dict_const()
+                record[self.toolname]['@idx'] = idx
+                record[self.toolname]['gauge'] = gauge = _dict_const()
+                record[self.toolname]['rate'] = rate = _dict_const()
+            elif ts_orig is None:
+                # We have not encountered a timestamp yet, so ignore
+                # all lines until the first timestamp.
+                continue
+            else:
+                key, value = line.strip().split(' ')
+                parts = key.split('_', 1)
+                if len(parts) == 1:
+                    if remaps is not None:
```

Comment:
What is going on with `remaps`? I don't understand.

Reply:
I've added some comments to help explain this a bit.